packages/appkit/src/client.ts

```diff
@@ -1543,11 +1549,31 @@ export class AppKit {
     this.setStatus('connected', chainNamespace)
 
     if (chainIdToUse && chainNamespace === activeNamespace) {
-      const caipNetwork = this.caipNetworks?.find(n => n.id.toString() === chainIdToUse.toString())
-      const fallBackCaipNetwork = this.caipNetworks?.find(n => n.chainNamespace === chainNamespace)
-      this.setCaipNetwork(caipNetwork || fallBackCaipNetwork)
+      let caipNetwork = this.caipNetworks?.find(n => n.id.toString() === chainIdToUse.toString())
```

Comment:
We we need to parse id's to string? Isn't safer to not do it? On top as I see `chainIdToUse` is coming from `params` which it should be number for EVM already - just to double check

Preferably we should avoid type casting IMO, wdyt?

Reply:
They are both `string | number` ! `syncAccount` could be triggered from for example wagmi which would set `chainId` to a number and this condition would not match cause our internal caipnetworks are strings